h2o-algos/src/main/java/hex/glm/ComputationState.java

```diff
@@ -403,13 +425,84 @@ private static void fillSubRange(int N, int c, int [] ids, double [] src, double
+   }
+
+  public static double[] expandToFullArray(double[] shortenArr, int[] activeCols, int _totalBetaLength, int nclasses, 
+                                           int betaLengthPerClass) {
+    if (activeCols == null)
+      return shortenArr;
+    int[] activeColsAllClass = genActiveColsAllClass(activeCols.length*nclasses,
+            betaLengthPerClass, activeCols, nclasses);
+    double[] fullArray = new double[_totalBetaLength];
+    fillSubRange(_totalBetaLength, 0, activeColsAllClass, shortenArr, fullArray);
+    return fullArray;
+  }
 
-  public double [] betaMultinomial(int c, double [] beta) {return extractSubRange(_activeData.fullN()+1,c,_activeDataMultinomial[c].activeCols(),beta);}
+  public static int[] genActiveColsAllClass(int activeColsLen, int numBetaPerClass, int[] activeColsOrig, int nclasses) {
+    int[] activeCols = new int[activeColsLen];
+    int offset = 0;
+    int[] activeColsOneClass = activeColsOrig;
+    for (int classIndex=0; classIndex < nclasses; classIndex++) {
+      int finalOffset = numBetaPerClass*classIndex;
+      int[] activeCols1Class = IntStream.of(activeColsOneClass).map(i->i+finalOffset).toArray();
+      int num2Copy = activeColsOneClass.length;
+      System.arraycopy(activeCols1Class, 0, activeCols, offset, num2Copy);
+      offset += num2Copy;
+    }
+    return activeCols;
+  }
+  
+  public int[] genActiveColsIndClass(int activeColsLen, int numBetaPerClass, int[] activeColsOrig, int activeClass,
+                                     int nclasses) {
+    int[] activeCols = new int[activeColsLen];// total length
+    int offset = 0;
+    int[] activeColsOneClass = activeColsOrig;
+    for (int classIndex = 0; classIndex < activeClass; classIndex++) {
+      int finalOffset = numBetaPerClass*classIndex;
+      int num2Copy = activeColsOneClass.length;
+      int[] activeCols1Class = IntStream.of(activeColsOneClass).map(i->i+finalOffset).toArray();
+      System.arraycopy(activeCols1Class, 0, activeCols, offset, num2Copy);
+      offset += num2Copy;
+    }
+    for (int classInd = activeClass; classInd < nclasses; classInd++) {
+      int finalOffset = numBetaPerClass*classInd;
+      int[] activeCols1Class = IntStream.range(0, numBetaPerClass).map(i->i+finalOffset).toArray();
+      System.arraycopy(activeCols1Class, 0, activeCols, offset, numBetaPerClass);
+      offset += numBetaPerClass;
+    }
+    return activeCols;
+  }
 
   public GLMSubsetGinfo ginfoMultinomial(int c) {
     return new GLMSubsetGinfo(_ginfo,(_activeData.fullN()+1),c,_activeDataMultinomial[c].activeCols());
   }
 
+  public GLMSubsetGinfo ginfoMultinomialRCC(int c) {
+    if (_activeData.fullN() + 1 == _activeData.activeCols().length)
+      return new GLMSubsetGinfo(_ginfo, (_activeData.fullN() + 1), c, IntStream.range(0, 
+              _activeData.activeCols().length).toArray());
+    else
+      return new GLMSubsetGinfo(_ginfo, (_activeData.fullN() + 1), c, _activeData.activeCols());
```

Comment:
Seems to me that we can keep just the `return new GLMSubsetGinfo(_ginfo, (_activeData.fullN() + 1), c, _activeData.activeCols());`. I assume if we have `_activeData.fullN() + 1 == _activeData.activeCols().length` then `IntStream.range(0, _activeData.activeCols().length).toArray() == _activeData.activeCols()`. 

Or is there some hidden reason why it wouldn't work?

Reply:
Tried it and found that active cols could be missing some values even if `_activeData.fullN() + 1 == _activeData.activeCols().length` so my hypothesis was false.